client_side/.infrastructure/infrastructure.sh

```diff
@@ -0,0 +1,201 @@
+
+  SECONDS=0
+  time_elapsed=0
+}
+
+# "Uninstalls" Bash Preexec by removing its triggers.
+# Remove all variable files created by the infrastructure
+# Stops the experiment completely by returning from the sourced scripts.
+end_experiment() {
+  # This effectively uninstalls Bash Pre-exec
+  # Makes it so any commands typed after the experiment has ended will not be
+  # passed through preexec and precmd.
+  PROMPT_COMMAND=${PROMPT_COMMAND_OG}
+  trap - DEBUG
+
+  # Remove all variable files
+  rm -r "${INFRA_DIR}/.*"
+  cd "${EXP_DIR}"
+
+  echo "Congratulations! You have completed the interactive portion of the" \
+    "experiment!"
+  echo "Please take some time to fill out the survey at ${SURVEY_URL} ."
+
+  return 0
+}
+
+# Creates the file system directory that the user will be working on.
+# If the directory already exists, removes it and creates it again.
+# Saves the user's current working directory and returnst to it once the
+# directory is set up.
+make_fs() {
+  pushd "${INFRA_DIR}" &> /dev/null
+
+  rm -rf "${FS_DIR}"
+  mkdir "${FS_DIR}" > /dev/null
+  tar -xzf "${INFRA_DIR}/fs.tgz" -C "${FS_DIR}"
+
+  # Moves the user to this directory
+  popd &> /dev/null
+}
+
+# Prints the list of resources that the user is allowed to use based on the
+# current treatment
+print_treatment() {
+  echo -n "For this half of the experiment you can use any online resources," \
+    "man pages, "
+  if [[ "$(cat "${INFRA_DIR}/.treatment")" == "T" ]]; then
+    echo "and Tellina <URL> to help you solve the tasks."
+  else
+    echo "but you can't use Tellina to help you solve the tasks."
+  fi
+}
+
+# Prints the current task number and its description.
+print_task() {
+  local task_code=$(cat "${INFRA_DIR}/.task_code")
+
+  echo "-----------------------------------------------------------------------"
+  echo "Task: ${curr_task}/${TASKS_SIZE}"
+
+  jq -r '.description' "${TASKS_DIR}/task_${task_code}/task_${task_code}.json"
+}
+
+# See documentation for ./verify_task.py for more details on what it does
+# Captures the stdout of verify_task.py into $status
+# Captures the exit code of verify_task.py into $EXIT
+verify_task() {
+  # Verify the output of the previous command.
+  local task_code="$(cat "${INFRA_DIR}/.task_code")"
+  local user_command="$(cat "${INFRA_DIR}/.command")"
+
+  status=$("${INFRA_DIR}"/verify_task.py ${task_code} ${user_command})
+  EXIT=$?
+}
+
+# Increment the current task number and update the true task code in .task_code
+# accordingly
+#   - Check if all the tasks are complete, if it is then skip the following
+#     steps and clean up instead.
+#   - If this current_task is equal to `TASKS_SIZE / 2`, switch treatments and
+#     task sets and notify the user.
+# Write "start task" to `.command`.
+# Writes to the log
+# Set time_elapsed=0, status="incomplete", SECONDS=0
+# Prints out the task's information
+next_task() {
+  make_fs
+
+  # Increment the number of tasks finished by the user
+  curr_task=$(( curr_task + 1 ))
```

Comment:
The variable name doesn't indicate whether it is a task number, task code, or something else.  I would call it task_num instead.

Reply:
Resolved in 5381c5b.

Reply:
You don't have to respond to every comment individually.  If you change the line the comment is on, I believe GitHub will automatically not show the comments in the "Files changed" tab.